src/cocotb/decorators.py

```diff
@@ -76,9 +83,16 @@ def wrapper(*args, **kwargs):
 def external(func: Callable[..., Result]) -> Callable[..., Coroutine[Any, Any, Result]]:
     """Decorator that turns a blocking function into a :term:`coroutine function`.
 
-    This turns a normal function that isn't a coroutine into a blocking coroutine.
-    Currently, this creates a new execution thread for each function that is
-    called.
+    When the coroutine function is called, it creates a coroutine that will block until the function completes.
+    Currently, this creates a new execution thread for each function that is called.
+    The coroutine can be directly :keyword:`await`\ ed or be put into a new :class:`~cocotb.task.Task`.
+    When the function completes, the result of the function will be returned from the coroutine.
```

Comment:
```suggestion
    When an external function call is ``await``\ ed, it creates a coroutine that will block until the wrapped function completes.
    Currently, this creates a new execution thread for each function that is called.
    The coroutine can be directly :keyword:`await`\ ed or be constructed into a :class:`~cocotb.task.Task`.
    When the wrapped function completes, the result of the function will be returned from the coroutine.
```

Reply:
> `When an external function call is ``await``\ ed,`

`external` just returns a coroutine function, so it needs to be called to create the coroutine, which can then be awaited or turned into a Task. I was trying to be explicit. Python's behavior is not obvious, probably especially to HDL engineers, but I do think it's more informative than the old docstring.

Reply:
In `test_external` you have things like `value = await external(calls_cocotb_function)(dut)`, which I think is difficult to understand in general.

Reply:
Something needs to be updated because `When the coroutine function is called, it creates a coroutine` sounds wrong.